src/sentry/eventstore/processing/multiredis.py

```diff
@@ -0,0 +1,72 @@
+import hashlib
+from datetime import timedelta
+from typing import TypeVar
+
+from rediscluster import RedisCluster
+
+from sentry import options
+from sentry.utils.codecs import JSONCodec
+from sentry.utils.kvstore.encoding import KVStorageCodecWrapper
+from sentry.utils.kvstore.redis import RedisKVStorage
+from sentry.utils.redis import redis_clusters
+
+from .base import EventProcessingStore
+
+T = TypeVar("T", str, bytes)
+
+
+class MultiRedisProcessingStore(EventProcessingStore):
+    """
+    Adapter to shift traffic from one redis cluster to another
+    """
+
+    def __init__(self, **options):
+        inner = MultiRedisKVStorage(
+            old_cluster=redis_clusters.get(options["old_cluster"]),
+            new_cluster=redis_clusters.get(options["new_cluster"]),
+        )
+        super().__init__(KVStorageCodecWrapper(inner, JSONCodec()))
+
+
+class MultiRedisKVStorage(RedisKVStorage[T]):
+    def __init__(self, old_cluster: RedisCluster, new_cluster: RedisCluster) -> None:
+        self.old_cluster = old_cluster
+        self.new_cluster = new_cluster
+
+    def use_new(self, key: bytes):
+        rollout = options.get("eventstore.processing.rollout")
+        intkey = int(hashlib.md5(key).hexdigest(), base=16)
+        return (intkey % 10000) / 10000 <= rollout
```

Comment:
Assuming you expect rollout to be <= 1.0 and >= 0.0 where 0.0 is off and 1.0 means 100%
Isn't this supposed to be `<`
if rollout is 0.0 this can return true now as `(intkey % 10000) / 10000` can be 0
if rollout is 1.0 this will not correspond to 100%

Reply:
Oops, yes the operator should be `<`.  I ran some local tests to make sure that 1.0 will move all keys to the new cluster.

```python
        import hashlib
        winners = 0
        for _ in range(0, 10000):
            intkey = int(hashlib.md5(uuid4().hex.encode("utf8")).hexdigest(), base=16)
            if (intkey % 10000) / 10000 < 1.0:
                winners += 1
        assert winners == 10000
```